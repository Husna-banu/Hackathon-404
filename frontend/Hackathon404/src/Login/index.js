import React, {useState, useEffect} from 'react';
import {
  View,
  Text,
  TextInput,
  StatusBar,
  SafeAreaView,
  Button,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';
import styles from './style';
import commonStyle from '../commonStyles';
import {getFetch, postFetch} from '../utils/fetchAPI';

export default function Login({navigation}) {
  const [stateData, setStateData] = useState({
    emailId: '',
    password: '',
    hotelName: '',
    hotelId: 0,
    listOfServices: [],
    passwordError: false,
    emailIdError: false,
    loginError: false,
  });

  useEffect(() => {
    getFetch('http://backendproject5.herokuapp.com/fetchHotelDetails')
      .then(lists => {
        console.log(lists);
        lists.forEach(list => {
          if (list.hotelId === 1234) {
            setStateData(state => ({
              ...state,
              hotelName: list.hotemName,
              hotelId: list.hotelId,
              listOfServices: list.listOfServices,
            }));
          }
        });
      })
      .catch(err => console.log(err));
  }, []);

  const onChangeText = (text, name) => {
    const nameField = `${name}Error`;
    if (text === '') {
      setStateData(state => ({
        ...state,
        [nameField]: true,
      }));
    } else {
      setStateData(state => ({
        ...state,
        [nameField]: false,
      }));
    }
    setStateData(state => ({...state, [name]: text}));
  };

  const _login = () => {
    if (stateData.emailId === '') {
      setStateData(state => ({
        ...state,
        emailIdError: true,
      }));
    } else if (stateData.password === '') {
      setStateData(state => ({
        ...state,
        passwordError: true,
      }));
    } else {
      const body = {
        userId: stateData.emailId,
        password: stateData.password,
      };

      postFetch(
        'http://backendproject5.herokuapp.com/fetchUserDetailsById',
        body,
      ).then(response => {
        console.log('response', response);
        if (response.status === 'Failed') {
          setStateData(state => ({
            ...state,
            loginError: true,
          }));
        } else {
          switch (response.userType) {
            case 'HOTEL_ADMIN':
              navigation.navigate('AdminDashboard', {
                hotelId: stateData.hotelId,
                listOfServices: stateData.listOfServices,
              });
              break;
            case 'GUEST':
              navigation.navigate('Dashborad', {
                hotelId: stateData.hotelId,
                listOfServices: stateData.listOfServices,
              });
              break;
            case 'SUPER_ADMIN':
              navigation.navigate('Dashborad', {
                hotelId: stateData.hotelId,
                listOfServices: stateData.listOfServices,
              });
              break;
          }
        }
      });
    }
  };
  const backToPage = () => {
    navigation.goBack();
  };
  return (
    <View style={commonStyle.container}>
      <StatusBar barStyle="dark-content" backgroundColor="skyblue" />
      <SafeAreaView>
        <View style={commonStyle.header}>
          <Icon
            name="arrow-left"
            style={commonStyle.backButton}
            size={20}
            onPress={backToPage}
          />
          <Text style={commonStyle.heading}>Login</Text>
        </View>
        <View style={commonStyle.content}>
          <View style={styles.emailContainer}>
            <Text style={styles.inputTitle}>Email address/Username: </Text>
            <TextInput
              style={styles.textInput}
              onChangeText={text => onChangeText(text, 'emailId')}
              value={stateData.emailId}
              placeholder="Email Address/User name"
            />
            {stateData.emailIdError && (
              <Text style={styles.errorMsg}>Please enter email address</Text>
            )}
          </View>
          <View style={styles.emailContainer}>
            <Text style={styles.inputTitle}>Password: </Text>
            <TextInput
              style={styles.textInput}
              onChangeText={text => onChangeText(text, 'password')}
              value={stateData.password}
              placeholder="Password"
              secureTextEntry={true}
            />
            {stateData.passwordError && (
              <Text style={styles.errorMsg}>Please enter password address</Text>
            )}
          </View>
          {stateData.loginError && (
            <Text style={styles.errorMsg}>
              Please enter valid email address/password
            </Text>
          )}
          <Button
            title="Login"
            onPress={() => _login()}
            style={{paddingTop: 10}}
          />
        </View>
      </SafeAreaView>
    </View>
  );
}
