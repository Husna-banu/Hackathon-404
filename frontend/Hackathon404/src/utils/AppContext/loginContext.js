import React from 'react'
const accessStateContext = React.createContext()
const accessDispatchContext = React.createContext()

function AccessReducer(state, action) {
  switch (action.type) {
    case 'login': {
      return { loginName: action.payload, login: true }
    }
    case 'logout': {
      return { loginName: action.payload, login: false }
    }
    default: {
      throw new Error(`Unhandled action type: ${action.type}`)
    }
  }
}

function AccessProvider({ children }) {
  const [state, dispatch] = React.useReducer(AccessReducer, { loginName: '', login: false })
  return (
    <accessStateContext.Provider value={state}>
      <accessDispatchContext.Provider value={dispatch}>
        {children}
      </accessDispatchContext.Provider>
    </accessStateContext.Provider>
  )
}

function useAccessState() {
  const context = React.useContext(accessStateContext)
  if (context === undefined) {
    throw new Error('useAccessState must be used within a AccessProvider')
  }
  return context
}

function useAccessDispatch() {
  const context = React.useContext(accessDispatchContext)
  if (context === undefined) {
    throw new Error('useAccessDispatch must be used within a AccessProvider')
  }
  return context
}

export { AccessProvider, useAccessState, useAccessDispatch }
