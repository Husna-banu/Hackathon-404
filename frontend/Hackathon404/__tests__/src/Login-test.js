import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import Login from '../../src/Login';
import {AccessProvider} from '../../src/utils/AppContext/loginContext';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
fetch = jest.fn(() => Promise.resolve());
const props = {
  navigate: jest.fn(),
  route: {
    params: {},
  },
};

it('renders correctly', async () => {
  const container = renderer.create(
    <AccessProvider>
      <Login {...props} />
    </AccessProvider>,
  );
  expect(container).toMatchSnapshot();
});
