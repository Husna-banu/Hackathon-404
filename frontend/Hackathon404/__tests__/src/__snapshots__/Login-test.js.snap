// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
      "top": 40,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "skyblue",
          "flexDirection": "row",
          "height": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "fontSize": 20,
            "fontWeight": "bold",
            "justifyContent": "center",
            "paddingLeft": 10,
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "paddingBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "paddingBottom": 10,
            }
          }
        >
          Email address/Username: 
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Email Address/User name"
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "gray",
              "borderWidth": 1,
              "height": 40,
              "padding": 5,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "paddingBottom": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "paddingBottom": 10,
            }
          }
        >
          Password: 
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "gray",
              "borderWidth": 1,
              "height": 40,
              "padding": 5,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
